xapian-core/tests/api_weight.cc

```diff
@@ -505,6 +505,86 @@ DEFINE_TESTCASE(pl2weight3, backend) {
     return true;
 }
 
+// Test exception for junk after serialised weight.
+DEFINE_TESTCASE(pl2plusweight1, !backend) {
+    Xapian::PL2PlusWeight wt(2.0, 0.9);
+    try {
+	Xapian::PL2PlusWeight b;
+	Xapian::PL2PlusWeight * b2 = b.unserialise(wt.serialise() + "X");
+	// Make sure we actually use the weight.
+	bool empty = b2->name().empty();
+	delete b2;
+	if (empty)
+	    FAIL_TEST("Serialised PL2PlusWeight with junk appended unserialised to empty name!");
+	FAIL_TEST("Serialised PL2PlusWeight with junk appended unserialised OK");
+    } catch (const Xapian::SerialisationError &) {
+
+    }
+    return true;
+}
+
+// Test for invalid values of c.
+DEFINE_TESTCASE(pl2plusweight2, !backend) {
```

Comment:
@samuelharden I wonder if grouping the test cases that test invalid argument exception together should serve our purpose here?
> Moreover, since we are writing new test we should give relevant names to the test

There seems to be a common practice of naming test cases as `<weighting-scheme><number>` for weighting schemes. So, I chose to stick with the convention for naming new test cases.